### PR TITLE
Add examples for Bitbucket Cloud and Update doc

### PR DESCRIPTION
**Pull request by savitaashture**
# Changes
* Add example for Bitbucket Cloud
* Update example of `Bitbucket` to `Bitbucket Server` because the current Bitbucket example supported by Triggers is for Bitbucket Server
* Update doc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Add example support for Bitbucket Cloud
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/approve
Are examples in v1alpha1 needed? I don't think we need to add examples to v1alpha1.

File: docs/interceptors.md [hunk 1]
```diff
@@ -242,24 +246,26 @@ spec:
...
 
-A Bitbucket `Interceptor` contains logic that validates and filters [Bitbucket server](https://confluence.atlassian.com/bitbucketserver) webhooks.
-It can validate the webhook's origin as described in [Webhooks](https://docs.gitlab.com/ee/user/project/integrations/webhooks.html). It does
-not support [Bitbucket cloud](https://support.atlassian.com/bitbucket-cloud/) due to lack of support for validating secrets; if you need this
-functionality with Bitbucket, use a [CEL `Interceptor`](#cel-interceptors) instead. The Bitbucket `Interceptor` always preserves  the payload
+Bitbucket interceptors has support for both Bitbucket server and Bitbucket cloud.
+
+### Bitbucket Server
+
+A Bitbucket Server `Interceptor` contains logic that validates and filters [Bitbucket server](https://confluence.atlassian.com/bitbucketserver) webhooks.
```
Comment (dibyom) [inline docs/interceptors.md:253]: Thanks @savitaashture I think separating out the difference between server and cloud is great. One minor thing is that  from the docs it seems like we have two separate interceptors. We only have one bitbucket interceptor which supports secret validation and event filtering for bitbucket server and only event filtering for bitbucket cloud

File: docs/interceptors.md [hunk 2]
```diff
@@ -282,26 +288,73 @@ For reference, below is an example legacy Bitbucket `Interceptor` definition:
...
+        - ref: bitbucket-server-binding
+      template:
+        ref: bitbucket-server-template
+```
+
+### Bitbucket Cloud
+
+A [Bitbucket cloud](https://support.atlassian.com/bitbucket-cloud/) is free to use and explore functionality, To know more about Bitbucket Cloud [follow](https://support.atlassian.com/bitbucket-cloud/docs/get-started-with-bitbucket-cloud/).
+Bitbucket Cloud doesn't support secret validation; hence its not secure compare to Bitbucket Server.
+[CEL `Interceptor`](#cel-interceptors) also used to get the similar functionality of Bitbucket Cloud.
```
Comment (dibyom) [inline docs/interceptors.md:317]: I think we can remove this line.

File: docs/interceptors.md [hunk 3]
```diff
@@ -282,26 +288,73 @@ For reference, below is an example legacy Bitbucket `Interceptor` definition:
...
-        - ref: bitbucket-binding
+        - ref: bitbucket-server-binding
+      template:
+        ref: bitbucket-server-template
+```
+
+### Bitbucket Cloud
+
+A [Bitbucket cloud](https://support.atlassian.com/bitbucket-cloud/) is free to use and explore functionality, To know more about Bitbucket Cloud [follow](https://support.atlassian.com/bitbucket-cloud/docs/get-started-with-bitbucket-cloud/).
+Bitbucket Cloud doesn't support secret validation; hence its not secure compare to Bitbucket Server.
```
Comment (dibyom) [inline docs/interceptors.md:316]:
```suggestion
At the moment, Bitbucket cloud does not support validating webhook payloads using shared secrets. See the [Secure Webhooks section](https://support.atlassian.com/bitbucket-cloud/docs/manage-webhooks/) of Bitbucket cloud docs for more information on securing your bitbucket webhooks
```

File: docs/interceptors.md [hunk 4]
```diff
@@ -282,26 +288,73 @@ For reference, below is an example legacy Bitbucket `Interceptor` definition:
...
       bindings:
-        - ref: bitbucket-binding
+        - ref: bitbucket-server-binding
+      template:
+        ref: bitbucket-server-template
+```
+
+### Bitbucket Cloud
+
+A [Bitbucket cloud](https://support.atlassian.com/bitbucket-cloud/) is free to use and explore functionality, To know more about Bitbucket Cloud [follow](https://support.atlassian.com/bitbucket-cloud/docs/get-started-with-bitbucket-cloud/).
```
Comment (dibyom) [inline docs/interceptors.md:315]: I don't think we need to describe what Bitbucket cloud is given that we do not do it for other providers.

File: docs/interceptors.md (hunk 3 above)
Comment (savitaashture) [reply docs/interceptors.md]: Updated 
Thank you

File: docs/interceptors.md (hunk 1 above)
Comment (savitaashture) [reply docs/interceptors.md]: Ye looks like that
Updated now
PTAL

**Comment by savitaashture** [comment]
> /approve Are examples in v1alpha1 needed? I don't think we need to add examples to v1alpha1.

@khrm removed `bitbucket-cloud` from v1alpha1 and kept bitbucket-server as it is because earlier we had `bitbucket` related example in `v1alpha1`

**Review by dibyom — APPROVED** [review]
/lgtm

**Pull request merged**